src/xdist/plugin.py

```diff
@@ -214,3 +215,12 @@ def worker_id(request):
         return request.config.workerinput["workerid"]
     else:
         return "master"
+
+
+@pytest.fixture(scope="session")
+def session_uuid(request):
+    """Return the uuid of the current test."""
```

Comment:
is there a reason for not returning a actual uuid instance?

Reply:
Perhaps I should rather name it `session_uid` as opposed to `session_uuid` so as to not give off the impression that this is indeed supposed to be a UUID object. I mean, it's implemented via UUID but the general idea is to just provide a string you can trust to be unique and shared amongst the workers. The idea was never to actually provide a UUID object.

I think a UUID object is only more annoying to work with no real benefit. However, I'm open to your perspective on this.

Reply:
Thanks for the clarification

I wonder if it should default to a textural id and allow setting the I'd from the cli as well

That way interesting ids matching ci systems could be supplied

I wonder if this would have a better place in pytest core or a pytest-testrun-id plugin to try things out

Btw, the name session Id is structurally mismatched as each subprocess is a own session within the same testrun

Reply:
> I wonder if it should default to a textural id and allow setting the I'd from the cli as well

Good idea, I'll add that.

> I wonder if this would have a better place in pytest core or a pytest-testrun-id plugin to try things out

I think due to the nature of the workers having to own the uids themselves, we'll have to do changes inside of pytest-xdist anyway. I'd suggest starting with that for the time being. If we eventually decide to make pytest itself also provide a uid, we can easily change our xdist part to fetch the uid from pytest instead of generating it on the spot. The rest of the changes would stay the way they currently are in this PR.

> Btw, the name session Id is structurally mismatched as each subprocess is a own session within the same testrun

Oh, I didn't know that. What name do you think would be better suited? `testrun_uid`?